src/NServiceBus.Core/Transports/TransportSendingConfigurationResult.cs

```diff
@@ -0,0 +1,26 @@
+namespace NServiceBus.Transports
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Represents the result for configuring the transport for sending.
+    /// </summary>
+    public class TransportSendingConfigurationResult
+    {
+        /// <summary>
+        /// Creates new result object.
+        /// </summary>
+        public TransportSendingConfigurationResult(Func<IDispatchMessages> dispatcherFactory,
+            Func<Task> preStartupCheck)
```

Comment:
Would it make sense to change preStartupCheck to be Task<bool> or make it somehow more explicit when the check fails? Now exception was to be thrown to stop transport startup which might not be obvious at first.